tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -49,6 +74,63 @@ public static JavaParser getJavaParser() {
+     * "ResolvedMethodDeclaration".
+     *
+     * @param jpClass the declaring class
+     * @param jpCallable the method in which the type is used
+     * @param methodArgs the arguments of the method
+     * @param expression a Java expression (e.g. "jpClass.getMethods()")
+     * @return the resolved type of the expression
+     * @throws ResolvedTypeNotFound if the type is not found
+     */
+    public static ResolvedType getResolvedTypeOfExpression(
+            TypeDeclaration<?> jpClass,
+            CallableDeclaration<?> jpCallable,
+            List<Triplet<String, String, String>> methodArgs,
+            String expression
+    ) throws ResolvedTypeNotFound {
+        String SYNTHETIC_METHOD_NAME = "__tratto__auxiliaryMethod";
+        if (jpClass instanceof ClassOrInterfaceDeclaration) {
+            BlockStmt syntheticMethodBody = jpClass.addMethod(SYNTHETIC_METHOD_NAME).getBody().get();
+            // add statement per method argument.
+            for (Triplet<String, String, String> methodArg : methodArgs) {
+                syntheticMethodBody.addStatement(methodArg.getValue2() + " " + methodArg.getValue0() + ";");
+            }
+            // get method declaration.
+            if (!jpCallable.getNameAsString().equals(jpClass.getNameAsString())) {
+                String jpMethodType = ((MethodDeclaration) jpCallable).getType().asString();
+                if (!jpMethodType.equals("void")) {
+                    syntheticMethodBody.addStatement(
+                            jpMethodType + " methodResultID = " + jpCallable.getNameAsString() + "(" +
+                                    methodArgs
+                                            .stream()
+                                            .map(Triplet::getValue0)
+                                            .collect(Collectors.joining(", "))
+                                    + ");"
+                    );
+                }
+                syntheticMethodBody.addStatement("var returnType = " + expression + ";");
+                return jpClass.asClassOrInterfaceDeclaration()
+                        .getMethodsByName(SYNTHETIC_METHOD_NAME).get(0)
+                        .getBody().get()
+                        .getStatements().getLast().get()
+                        .asExpressionStmt().getExpression()
+                        .asVariableDeclarationExpr().getVariables().get(0)
+                        .getInitializer().get()
+                        .calculateResolvedType();
+            }
+        }
+        throw new ResolvedTypeNotFound(String.format(
+                "ResolvedType of expression %s of class %s and method %s not found.",
+                expression,
+                jpClass.getNameAsString(),
+                jpCallable.getNameAsString()
+        ));
+    }
+
     /**
      * Given a syntactically valid Java expression, evaluates its return type, including package and class.
```

Comment:
Simply put, it would be doing something like this:
```java
val someVariable = <java_expression_here>;
```
where the output would be the package and class of `someVariable`.

Reply:
OK, thanks for the explanation.  You mean "determine's the expression's type".  There is no evaluation occurring, and expressions do not have a return type (though they do have a type).